src/Components/Components/src/EventCallbackFactoryBinderExtensions.cs

```diff
@@ -577,6 +632,23 @@ public static bool ConvertCore(object obj, out T value)
                 }
             };
             return factory.Create<UIChangeEventArgs>(receiver, callback);
+
+            static DateTime ConvertDateTime(object obj, string format)
+            {
+                var text = (string)obj;
+                if (string.IsNullOrEmpty(text))
+                {
+                    return default;
+                }
+                else if (format != null && DateTime.TryParseExact(text, format, CultureInfo.InvariantCulture, DateTimeStyles.RoundtripKind, out var value))
```

Comment:
Why `InvariantCulture` for dates but `CurrentCulture` for (e.g.) numbers?

Reply:
This is the existing code for dates with formats. I'm not going to message with the existing beahvior because we have another issue for that.